src/types/module.rs

```diff
@@ -135,49 +217,52 @@ impl PyModule {
         self.setattr(name, value.into_py(self.py()))
     }
 
-    /// Adds a new extension type to the module.
-    ///
-    /// This is a convenience function that initializes the `class`,
-    /// sets `new_type.__module__` to this module's name,
-    /// and adds the type to this module.
-    pub fn add_class<T>(&self) -> PyResult<()>
-    where
-        T: PyClass,
-    {
-        self.add(T::NAME, <T as PyTypeObject>::type_object(self.py()))
-    }
-
-    /// Adds a function or a (sub)module to a module, using the functions __name__ as name.
+    /// Adds a new class to the module.
     ///
-    /// Use this together with the`#[pyfunction]` and [wrap_pyfunction!] or `#[pymodule]` and
-    /// [wrap_pymodule!].
+    /// Notice that this method does not take an argument.
+    /// Instead, this method is *generic*, and requires us to use the
+    /// "turbofish" syntax to specify the class we want to add.
     ///
+    /// # Examples
     /// ```rust
     /// use pyo3::prelude::*;
+    ///
+    /// #[pyclass]
+    /// struct Foo{ /* fields omitted */ }
+    ///
     /// #[pymodule]
-    /// fn utils(_py: Python, _module: &PyModule) -> PyResult<()> {
+    /// fn my_module(_py: Python, module: &PyModule) -> PyResult<()> {
+    ///     module.add_class::<Foo>()?;
     ///     Ok(())
     /// }
+    ///  ```
+    /// Python code can see this class as such:
+    /// ```python
+    /// from my_module import Foo
     ///
-    /// #[pyfunction]
-    /// fn double(x: usize) -> usize {
-    ///     x * 2
-    /// }
-    /// #[pymodule]
-    /// fn top_level(_py: Python, module: &PyModule) -> PyResult<()> {
-    ///     module.add_wrapped(pyo3::wrap_pymodule!(utils))?;
-    ///     module.add_wrapped(pyo3::wrap_pyfunction!(double))
-    /// }
+    /// print("Foo is", Foo)
     /// ```
-    ///
-    /// You can also add a function with a custom name using [add](PyModule::add):
-    ///
-    /// ```rust,ignore
-    /// m.add("also_double", wrap_pyfunction!(double)(m)?)?;
+    /// This will result in the following output:
+    /// ```text
+    /// Foo is <class 'builtins.Foo'>
     /// ```
+    /// Note that as we haven't defined a [constructor][1], Python code can't actually
+    /// make an *instance* of `Foo` (or *get* one for that matter, as we haven't exported
+    /// anything that can return instances of `Foo`).
     ///
-    /// **This function will be deprecated in the next release. Please use the specific
```

Comment:
I turned this into an actual `#[deprecated(...)]`.

I'm assuming this is still what we want...?

Reply:
I've removed the deprecation